return table.freeze({
	ImageDefaults = require(script.ImageDefaults),
	SkillAssetBindingDefinition = require(script.SkillAssetBindingDefinition),
	SkillAssetBindings = require(script.SkillAssetBindings),
	SkillAssetResolver = require(script.SkillAssetResolver),
	StudioSkillAssetOverrides = require(script.StudioSkillAssetOverrides),
	StatusAssetBindingDefinition = require(script.StatusAssetBindingDefinition),
	StatusAssetBindings = require(script.StatusAssetBindings),
	StatusAssetResolver = require(script.StatusAssetResolver),
	StudioStatusAssetOverrides = require(script.StudioStatusAssetOverrides),
})
